pantheon-bundle/frontend/src/app/productListing.tsx

```diff
@@ -1,188 +1,119 @@
-                              menuItems={[
-                                <OptionsMenuItem onSelect={this.onSelect(event, data)} key="dropdown">Product Details</OptionsMenuItem>]}
-                              isOpen={data.isOpen}
-                              toggle={<OptionsMenuToggle onToggle={this.onToggle(data["jcr:uuid"])} toggleTemplate={<CaretDownIcon aria-hidden="true" />} aria-label="Sort by" hideCaret={true} />} />
-                          </DataListAction>
-                        </DataListCell>
-                      ]}
-                    />
-                  </DataListItemRow>
-                </DataListItem>))}
-              {this.state.isEmptyResults && (
-                <DataListItem aria-labelledby="single-action-item0" data-testid="emptyResults">
-                  <DataListItemRow>
-                    <DataListItemCells
-                      dataListCells={[
-                        <DataListCell key="primary content" width={2}>
-                          <span className="sp-prop-nosort" id="product-name">No products found</span>
-                        </DataListCell>
-                      ]}
-                    />
-                  </DataListItemRow>
-                </DataListItem>
-              )}
-            </DataList>
-          </div>)}
-      </React.Fragment>
-    );
-  }
+  useEffect(() => {
+    setFilteredProducts(allProducts)
+  }, [allProducts])
 
-  private getProducts = (allProducts) => {
-    fetch(this.getProductsUrl())
-      .then(response => response.json())
-      .then(responseJSON => {
-        let key; let singleProduct;
-        for (const i of Object.keys(responseJSON.results)) {
-          key = Object.keys(responseJSON.results)[i];
-          singleProduct = responseJSON.results[key];
-          singleProduct = Object.assign({ "isOpen": false }, singleProduct)
-          allProducts.push(singleProduct)
-        }
-        this.setState({ results: allProducts })
-      })
-      .then(() => {
-        if (Object.keys(this.state.results).length === 0) {
-          this.setState({
-            isEmptyResults: true
-          });
-        } else {
-          this.setState({
-            isEmptyResults: false
-          });
-        }
-      })
-  }
-
-  private getProductsUrl() {
-    const backend = "/content/products.query.json?nodeType=pant:product&orderby=name"
-    return backend
-  }
+  const onToggle = (id) => {
+    updateProduct(id);
+  };
 
-  private onToggle = (id) => (event: any) => {
-    this.state.results.map(data => {
-      if (data["jcr:uuid"] === id) {
-        (data.isOpen as any) = !data.isOpen
-        this.setState({ isProductDetails: false })
-      }
-    });
+  const onSelect = (id) => {
+    // @ts-ignore
+    window.location += `/${id}`;
   };
 
-  private onSelect = (event, data) => () => {
-    this.setState({
-      isProductDetails: !this.state.isProductDetails,
-      productName: data.name,
+  const setInput = (text, event) => {
```

Comment:
I can has comment for function?

Reply:
Does this one need a comment? I vote yes, but if you feel like it doesn't I'm cool with it :D

Reply:
yes added a comment to this function in a diff commit